statics/workflows/flow-validation.yaml

```diff
@@ -92,38 +97,66 @@ Source: |
       capture = client.captures.create(capture)
       var timediff = interval * 1000;
       sleep(timediff)
-      var flowExpected = client.gremlin.G().V().Has('Captures.ID', capture.UUID, 'Type', NE('ovsport')).result();
-      var flowCaptured = client.gremlin.G().Flows().Has('CaptureID', capture.UUID).CaptureNode().Dedup().result();
-      var noflows = [];
 
-      if (protocol != "udp" && flowExpected.length && flowExpected.length == flowCaptured.length) {
-        flowCaptured = client.gremlin.G().Flows().Has('CaptureID', capture.UUID, 'Metric.ABPackets', GT(0), 'Metric.BAPackets', GT(0)).CaptureNode().Dedup().result();
+      var captureValidation = function(Expected) {
+        var noCapture = [];
+        for (var i = 0; i != Expected.length; i++) {
+          if (Expected[i].Metadata.State == "DOWN") {
+            noCapture.push(Expected[i]);
+          }
+        }
+        return noCapture
       }
 
-      if (flowExpected.length != flowCaptured.length) {
-        for (var i = 0; i != flowExpected.length; i++) {
+      var flowValidation = function(Expected, Captured) {
+        var noFlows = [];
+        for (var i = 0; i != Expected.length; i++) {
           var found = false;
 
-          for (var j = 0; j != flowCaptured.length; j++) {
-            if (flowExpected[i].ID === flowCaptured[j].ID) {
+          for (var j = 0; j != Captured.length; j++) {
+            if (Expected[i].ID === Captured[j].ID) {
               found = true;
               break;
             }
           }
 
           if (!found) {
-            noflows.push(flowExpected[i]);
+            noFlows.push(Expected[i]);
           }
         }
+        return noFlows
       }
 
-      if (!flowExpected.length) {
-        result["Error"] = "No node with capture"
-      } else if (noflows.length == 0) {
-        result["Connectivity"] = true
-      } else {
+      var noCapture = []
+      var flowExpected = client.gremlin.G().V().Has('Captures.ID', capture.UUID, 'Type', NE('ovsport')).result();
```

Comment:
does the following gremlin expression not enough to ignore `DOWN` interfaces ?

```
client.gremlin.G().V().Has('Captures.ID', capture.UUID, 'Type', NE('ovsport'), 'State', 'DOWN')
```

Reply:
Yes, but with this it will end up giving all other nodes followed by the node (with 'State : DOWN') without including the node (with 'State : DOWN').

Reply:
Not sure to understand your answer, but I'm even not sure we need to do something :)
I'm saying that because if the state is `DOWN` the capture won't start and the node won't have the `Capture.ID` field, meaning that we will get only the node being able to capture, those we want to check.